units/migrate_volume_job.go

```diff
@@ -0,0 +1,286 @@
+			return
+		}
+	}
+
+	// Terminate initial host
+	ts := utility.RoundPartOfMinute(1).Format(TSFormat)
+	terminateJob := NewSpawnHostTerminationJob(j.initialHost, evergreen.User, ts)
+	if err := j.env.RemoteQueue().Put(ctx, terminateJob); err != nil {
+		j.AddRetryableError(errors.Wrapf(err, "enqueueing initial host termination job '%s'", terminateJob.ID()))
+		return
+	}
+}
+
+// stopInitialHost inspects the initial host's status and stops it if the host is still running.
+func (j *volumeMigrationJob) stopInitialHost(ctx context.Context) {
+	// Stop host if this operation has not been attempted yet
+	if j.SpawnhostStopJobID == "" {
+		ts := utility.RoundPartOfMinute(1).Format(TSFormat)
+		stopJob := NewSpawnhostStopJob(j.initialHost, evergreen.User, ts)
+		err := j.env.RemoteQueue().Put(ctx, stopJob)
+		if err != nil {
+			j.AddRetryableError(err)
+			return
+		}
+		grip.Info(message.Fields{
+			"message":         "stopping initial host",
+			"job_id":          j.ID(),
+			"initial_host_id": j.InitialHostID,
+		})
+		j.SpawnhostStopJobID = stopJob.ID()
+		j.AddRetryableError(errors.Errorf("initial host '%s' not yet stopped", j.InitialHostID))
+		return
+	}
+
+	// Mark host as stopped
+	if j.initialHost.Status == evergreen.HostStopped {
+		grip.Info(message.Fields{
+			"message":         "initial host stopped",
+			"job_id":          j.ID(),
+			"initial_host_id": j.InitialHostID,
+		})
+		return
+	}
+
+	// If host is not stopped, verify whether the job is still in progress and error if it completed without stopping.
+	stopJob, foundJob := j.env.RemoteQueue().Get(ctx, j.SpawnhostStopJobID)
+	if !foundJob {
+		j.AddError(errors.Errorf("spawnhost-stop job '%s' could not be found", j.SpawnhostStopJobID))
+		return
+	}
+
+	if stopJob.Status().Completed {
+		j.AddError(errors.Errorf("host '%s' failed to stop", j.SpawnhostStopJobID))
+		return
+	}
+
+	j.AddRetryableError(errors.Errorf("initial host '%s' not yet stopped", j.InitialHostID))
+}
+
+// startNewHost attempts to start a new host with the volume attached.
+func (j *volumeMigrationJob) startNewHost(ctx context.Context) {
+	// Ensure volume has been detached
+	if j.volume.Host == j.InitialHostID {
+		j.AddRetryableError(errors.New("volume still attached to host"))
+		return
+	}
+
+	j.ModifyOptions.HomeVolumeID = j.VolumeID
+	intentHost, err := cloud.CreateSpawnHost(ctx, j.ModifyOptions, j.env.Settings())
+	if err != nil {
+		j.AddRetryableError(errors.Wrapf(err, "creating new intent host"))
+		return
+	}
+
+	if err := intentHost.Insert(); err != nil {
+		j.AddError(errors.Wrap(err, "inserting new intent host"))
+		return
+	}
+	grip.Info(message.Fields{
+		"message":        "new intent host created",
+		"job_id":         j.ID(),
+		"intent_host_id": intentHost.Id,
+	})
+	j.NewHostID = intentHost.Id
+}
+
+// finishJob marks the job as completed and attempts some additional cleanup if this is the job's final attempt.
+func (j *volumeMigrationJob) finishJob(ctx context.Context) {
```

Comment:
Should this job log either to Splunk or to the event log when it fails to migrate the volume?